superset-frontend/src/chart/chartAction.js

```diff
@@ -258,41 +331,31 @@ export function exploreJSON(
     const logStart = Logger.getTimestamp();
     const controller = new AbortController();
 
-    const useLegacyApi = shouldUseLegacyApi(formData);
-
-    let requestParams = {
+    const requestParams = {
       signal: controller.signal,
       timeout: timeout * 1000,
     };
+    if (dashboardId) requestParams.dashboard_id = dashboardId;
 
-    if (allowDomainSharding) {
-      requestParams = {
-        ...requestParams,
-        mode: 'cors',
-        credentials: 'include',
-      };
-    }
-
-    const queryPromiseRaw = useLegacyApi
-      ? legacyChartDataRequest(
-          formData,
-          force,
-          method,
-          dashboardId,
-          requestParams,
-        )
-      : v1ChartDataRequest(formData, force, requestParams);
+    const chartDataRequest = getChartDataRequest(
+      formData,
+      'json',
+      'full',
+      force,
+      method,
+      requestParams,
+    );
 
     dispatch(chartUpdateStarted(controller, formData, key));
 
-    const queryPromiseCaught = queryPromiseRaw
-      .then(({ json }) => {
+    const chartDataRequestCaught = chartDataRequest
+      .then(response => {
         // new API returns an object with an array of restults
-        // problem: json holds a list of results, when before we were just getting one result.
+        // problem: response holds a list of results, when before we were just getting one result.
         // `queryResponse` state is used all over the place.
```

Comment:
shoud this `queryResponse` be changed to `chartDataResponse` to keep things in step?

Reply:
Good catch! 👍